src/DIRAC/Core/Tornado/Server/TornadoREST.py

```diff
@@ -171,25 +171,12 @@ def _pre_initialize(cls) -> list:
         """
         urls = []
         # Look for methods that are exported
-        for mName in cls.__dict__:
-            mObj = cls.__dict__[mName]
-
-            if cls.METHOD_PREFIX and mName.startswith(cls.METHOD_PREFIX):
-                # Target methods begin with a prefix defined for all supported http methods,
-                # e.g.: def export_myMethod(self):
-                prefix = len(cls.METHOD_PREFIX)
-            elif _prefix := [
-                p for p in cls.SUPPORTED_METHODS if mName.startswith(f"{p.lower()}_")  # pylint: disable=no-member
-            ]:
-                # Target methods begin with the name of the http method,
-                # e.g.: def post_myMethod(self):
-                prefix = len(_prefix[-1]) + 1
-            else:
-                # The name of the target method must contain a special prefix
-                continue
-
-            # if the method exists we will continue
-            if callable(mObj) and (methodName := mName[prefix:]):
+        for prefix in [cls.METHOD_PREFIX] if cls.METHOD_PREFIX else cls.SUPPORTED_METHODS:
+            prefix = prefix.lower()
+            if prefix == "get_":
+                raise NotImplementedError("This is fundamentally broken as tornado has methods named get_")
+            for mName, mObj in inspect.getmembers(cls, lambda x: callable(x) and x.__name__.startswith(prefix)):
+                methodName = mName[len(prefix) :]
```

Comment:
IIUC `cls.SUPPORTED_METHODS` its a list of names without "_" (for ex.: ('GET', 'HEAD', 'POST', 'DELETE', 'PATCH', 'PUT', 'OPTIONS')), it means that `'GET'.lower() == "get_"` isn`t true. So it seems that exception will not work. 

I propose to leave the logic as it was, adding the exception of tornado methods. As I understand it, this will solve the purpose of this PR and will not break the existing logic.


```suggestion
        # Look for methods that are exported, except for original tornado methods, such as `get_secure_cookie`
        tornadoMethods = [name for name, cls_obj in inspect.getmembers(BaseRequestHandler, lambda x: callable(x))]

        for mName, mObj in [(name, cls_obj) for name, cls_obj in inspect.getmembers(cls, lambda x: callable(x)) if name not in tornadoMethods]:

            if cls.METHOD_PREFIX and mName.startswith(cls.METHOD_PREFIX):
                # Target methods begin with a prefix defined for all supported http methods,
                # e.g.: def export_myMethod(self):
                prefix = len(cls.METHOD_PREFIX)
            elif _prefix := [
                p for p in cls.SUPPORTED_METHODS if mName.startswith(f"{p.lower()}_")  # pylint: disable=no-member
            ]:
                # Target methods begin with the name of the http method,
                # e.g.: def post_myMethod(self):
                prefix = len(_prefix[-1]) + 1
            else:
                # The name of the target method must contain a special prefix
                continue

            # if the method exists we will continue
            if methodName := mName[prefix:]:
```

Reply:
@chrisburr any updates here?

Reply:
I don't think this is an acceptable solution and I think we should instead come up with a long-term solution for 8.1.